cmd/gokr-packer/packer.go

```diff
@@ -423,17 +457,48 @@ func logic() error {
 
 	partuuid := derivePartUUID(*hostname)
 	usePartuuid := true
+	var (
+		updateHttpClient         *http.Client
+		foundMatchingCertificate bool
+		updateBaseUrl            *url.URL
+	)
+
 	if *update != "" {
+		updateBaseUrl, err = url.Parse(*update)
+		if err != nil {
+			return err
+		}
+
+		updateHttpClient, foundMatchingCertificate, err = httpclient.GetTLSHttpClientByTLSFlag(useTLS, updateBaseUrl)
+		if err != nil {
+			return fmt.Errorf("gettting http client by tls flag: %v", err)
+		}
+		remoteScheme, err := httpclient.GetRemoteScheme(updateBaseUrl)
+		if remoteScheme == "https" {
+			updateBaseUrl.Scheme = "https"
+			*update = updateBaseUrl.String()
+		}
+
+		if updateBaseUrl.Scheme != "https" && foundMatchingCertificate {
+			fmt.Printf("\n")
+			fmt.Printf("!!!WARNING!!! Possible SSL-Stripping detected!\n")
+			fmt.Printf("Found certificate for hostname in your client configuration but the host does not offer https!\n")
+			fmt.Printf("\n")
+			if !*tlsInsecure {
+				return fmt.Errorf("canceled")
```

Comment:
Here and in the other file, let’s make this error message a little more descriptive. How about:

```
return fmt.Errorf("update canceled: TLS certificate found, but negotiating a TLS connection with the target failed")
```